### ISSUE TITLE
Problem with loss

### ISSUE THREAD

**Issue by khoaprogrammer** [experiencor/basic-yolo-keras#18]
I'm training a tiny yolo with 600 pic without loading pretrain weight. The problem is at the 1st epoch, the loss is very low (about 0.07) and the model was saved. At the 2nd epoch, the loss increase to 2 and after that it slowly descrease. The problem is the model was saved at the 1st epoch and after that it can't be saved anymore be cause the loss is alway higher than it is at the 1st epoch! please help

Epoch 1/50
38/39 [============================>.] - ETA: 20s - loss: 0.7124 Epoch 00000: val_loss improved from inf to 0.07960, saving model to best_weights.h5
39/39 [==============================] - 1103s - loss: 0.6963 - val_loss: 0.0796
Epoch 2/50
38/39 [============================>.] - ETA: 20s - loss: 0.9856 Epoch 00001: val_loss did not improve
39/39 [==============================] - 1093s - loss: 1.0072 - val_loss: 1.6463
Epoch 3/50
38/39 [============================>.] - ETA: 20s - loss: 1.4972 Epoch 00002: val_loss did not improve
39/39 [==============================] - 1096s - loss: 1.4875 - val_loss: 1.5270
Epoch 4/50
38/39 [============================>.] - ETA: 20s - loss: 1.2201 Epoch 00003: val_loss did not improve
39/39 [==============================] - 1100s - loss: 1.2121 - val_loss: 1.8843
Epoch 5/50
38/39 [============================>.] - ETA: 20s - loss: 1.0720 Epoch 00004: val_loss did not improve
39/39 [==============================] - 1091s - loss: 1.0613 - val_loss: 1.4738
Epoch 6/50
38/39 [============================>.] - ETA: 20s - loss: 0.9571 Epoch 00005: val_loss did not improve
39/39 [==============================] - 1085s - loss: 0.9489 - val_loss: 1.4657
Epoch 7/50
38/39 [============================>.] - ETA: 20s - loss: 0.8815 Epoch 00006: val_loss did not improve
39/39 [==============================] - 1083s - loss: 0.8774 - val_loss: 1.4764
Epoch 8/50
38/39 [============================>.] - ETA: 20s - loss: 0.7926 Epoch 00007: val_loss did not improve
39/39 [==============================] - 1085s - loss: 0.7873 - 

PS: I'm using my train set as my test set to test the model, is that ok???

**Comment by experiencor** [comment]
The loss starts low because of the warmup stage, which is defaulted to 100. It's advisable to leave it like that as warmup training is very important. Instead, you can start training the second time with the "pretrained_weights" in the config.json file set to "best_weights.h5". It will start from this weights.

It's not a good idea to use the train set as the test set. I think you should leave the test set empty. The code in train.py will automatically split the given train set to the train set and test set using the ration 0.8 if it cannot find the test set.
